core/unit_test/TestStringManipulation.cpp

```diff
@@ -176,7 +170,6 @@ KOKKOS_FUNCTION constexpr bool test_strncat() {
 }
 static_assert(test_strncat());
 
-#if !defined(KOKKOS_COMPILER_GNU) && (KOKKOS_COMPILER_GNU >= 540)
```

Comment:
This condition was unsatisfiable - again, probably should have been `||`

Reply:
Yes, I agree.

Reply:
My bad here too